kola/tests/misc/nvidia.go

```diff
@@ -3,50 +3,209 @@ package misc
+	// For amd64 the suffix is x86-64, for arm64 it's arm64
+	if kola.QEMUOptions.Board == "arm64-usr" {
+		params["ARCH_SUFFIX"] = "arm64"
+	} else {
+		params["ARCH_SUFFIX"] = "x86-64"
+	}
+
+	butane, err := testsutil.ExecTemplate(nvidiaOperatorTemplate, params)
+	if err != nil {
+		c.Fatalf("ExecTemplate: %s", err)
+	}
+
+	m, err := testsutil.NewMachineWithLargeDisk(c, "32G", conf.Butane(butane))
+	if err != nil {
+		c.Fatalf("Cluster.NewMachine: %s", err)
+	}
+
+	if err = waitForNvidiaDriver(&c, &m); err != nil {
+		c.Fatal(err)
+	}
+	_ = c.MustSSH(m, "sudo systemctl cat nvidia.service")
+	_ = c.MustSSH(m, "sudo systemd-sysext status")
+	c.AssertCmdOutputContains(m, "sudo systemd-sysext status", "nvidia_runtime")
+	c.AssertCmdOutputContains(m, "sudo systemd-sysext status", "nvidia-driver")
+	_ = c.MustSSH(m, `curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 \
+	&& chmod 700 get_helm.sh \
+	&& HELM_INSTALL_DIR=/opt/bin PATH=$PATH:/opt/bin ./get_helm.sh`)
+	_ = c.MustSSH(m, "sudo kubeadm init --pod-network-cidr=10.244.0.0/16")
+	_ = c.MustSSH(m, `mkdir -p $HOME/.kube
+	sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+	sudo chown $(id -u):$(id -g) $HOME/.kube/config`)
+	_ = c.MustSSH(m, "kubectl apply -f https://github.com/flannel-io/flannel/releases/latest/download/kube-flannel.yml")
+	_ = c.MustSSH(m, "kubectl taint nodes --all node-role.kubernetes.io/control-plane-")
+	_ = c.MustSSH(m, "kubectl describe nodes $HOSTNAME")
+	err = util.Retry(5, 10*time.Second, func() error {
+		out, err := c.SSH(m, "kubectl get nodes")
+		if err != nil {
+			return err
+		}
+		if strings.Contains(string(out), "NotReady") {
+			return fmt.Errorf("nodes not ready: %s", string(out))
+		}
+		return nil
+	})
+	if err != nil {
+		c.Fatalf("%v", err)
+	}
+	_ = c.MustSSH(m, "/opt/bin/helm repo add nvidia https://helm.ngc.nvidia.com/nvidia  && /opt/bin/helm repo update")
+	_ = c.MustSSH(m, `/opt/bin/helm install --wait --generate-name \
+	-n gpu-operator --create-namespace \
+	--version v24.9.2 \
+	nvidia/gpu-operator \
+	--set driver.enabled=false \
+	--set toolkit.enabled=false \
+	`)
+	_ = c.MustSSH(m, "/opt/bin/helm ls")
+	err = util.Retry(10, 10*time.Second, func() error {
+		out, err := c.SSH(m, "kubectl get pods --all-namespaces -o json | jq '.items[] | select(.status.phase != \"Running\" and .status.phase != \"Succeeded\") | .metadata.name'")
+		if err != nil {
+			return err
+		}
+		lines := strings.Split(string(out), "\n")
+		if len(lines) > 0 && lines[0] != "" {
+			return fmt.Errorf("pods not ready: %d: %v", len(lines), lines)
+		}
+		return nil
+	})
+	_ = c.MustSSH(m, "kubectl get pods --all-namespaces")
+	if err != nil {
+		c.Fatalf("%v", err)
+	}
+	_ = c.MustSSH(m, `kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: cuda-vectoradd
+spec:
+  restartPolicy: OnFailure
+  containers:
+  - name: cuda-vectoradd
+    image: "nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda11.7.1-ubuntu20.04"
```

Comment:
Maybe this docker tag could also be a constant?

Reply:
done